modules/handlers/command_handlers.py

```diff
@@ -290,3 +291,137 @@ def forwarded_post_msg(update: Update, context: CallbackContext):
 
 
 # endregion
+
+# region report spot
+
+def report_post(update: Update, context: CallbackContext) -> int:
+    """Handles the reply to the key "Report".
+    Checks the message the user wants to report, and goes to the final step
+
+    Args:
+        update (Update): update event
+        context (CallbackContext): context passed by the handler
+
+    Returns:
+        int: next state of the conversation
+    """
+    info = get_message_info(update, context)
+
+    if not check_message_type(update.message):  # the type is NOT supported
+        info['bot'].send_message(
+            chat_id=info['chat_id'],
+            text="Questo tipo di messaggio non è supportato\nÈ consentito solo testo, stikers, immagini, audio, video o poll\n\
+                Invia il post che vuoi pubblicare\nPuoi annullare il processo con /cancel")
+        return STATE['reporting_spot']
+    
+    chat_id = config_map['meme']['group_id'] # should be admin group
+    channel_id = config_map['meme']['channel_group_id'] # should be users group
+
+    abusive_message_id = MemeData.get_last_post_report(user_id=info['sender_id'], channel_id=channel_id)
+
+    info['bot'].forward_message(chat_id=chat_id,
+                                from_chat_id=channel_id,
+                                message_id=abusive_message_id)
+
+    info['bot'].sendMessage(chat_id=chat_id, text="🚨🚨 SEGNALAZIONE 🚨🚨\n\n" + info['text'])
+    info['bot'].send_message(chat_id=info['chat_id'],
+                            text="Gli admins verificheranno quanto accaduto. Grazie per la collaborazione!")
+
+    return STATE['end']
+
+# endregion
+
+# region report user
+
+def report_cmd(update: Update, context: CallbackContext) -> int:
+    """Handles the reply to the key "Report".
+    Checks the message the user wants to report
+
+    Args:
+        update (Update): update event
+        context (CallbackContext): context passed by the handler
+
+    Returns:
+        int: next state of the conversation
+    """
+    info = get_message_info(update, context)
+    if update.message.chat.type != "private":  # you can only post with a private message
+        info['bot'].send_message(
+            chat_id=info['chat_id'],
+            text="Non puoi usare quest comando ora\nChatta con @Spotted_DMI_bot in privato",
+        )
+        return STATE['end']
+
+    user_report = MemeData.get_last_user_report(user_id=info['sender_id'])
+    if user_report and user_report['is_sent']:
+        delta_time = datetime.now() - datetime.strptime(user_report['message_date'], "%Y-%m-%d %H:%M:%S.%f")
+        delta_minutes = delta_time.total_seconds()/60
+        remain_minutes = int(config_map['report_wait_mins'] - delta_minutes)
```

Comment:
The field **'report_wait_mins'** was not added in the _settings.yaml.dist_ file.
Personally I would put it in the 'meme' sub-category